tensordict/tensordict.py

```diff
@@ -1717,7 +1722,11 @@ def permute(
         )
 
     def __repr__(self) -> str:
+        if self._being_called:
```

Comment:
Same as `flatten_keys`, we can make this fault-tolerant with a `try / finally`

Reply:
> Regarding the tests, I'm happy with tests being independent of the others although I wonder if the fact that a call to `__eq__` is prohibited should not be addressed by this PR. TBH I did not quite get what the issue was. Can you elaborate a bit more? You can also code it and push the code with the tests breaking, that would give me some visibility on what's happening.

This issue was about infinite recursion during print, flatten_keys, and list(keys()). Issue # https://github.com/pytorch-labs/tensordict/issues/106. This is example of code how to verify this PR:
<details>
  <summary>Code</summary>

```python
tensordict = TensorDict({
     "key 1": torch.ones(3, 4, 5),
     "key 2": torch.zeros(3, 4, 5, dtype=torch.bool),
}, batch_size=[3, 4])

tensordict2 = TensorDict({
    "super key 1": torch.ones(3, 4, 6),
    "super key 2": torch.zeros(3, 4, 7, dtype=torch.bool),
}, batch_size=[3, 4])

tensordict["innerTD"] = tensordict2
tensordict2["self 2"] = tensordict2
tensordict["self"] = tensordict
tensordict["key 3"] = torch.zeros(3, 4, 7, dtype=torch.bool)

f_keys = tensordict.flatten_keys()  # Ok
print(f_keys)
td_list = list(tensordict.keys(include_nested=True))  # Ok
print(td_list)
print(tensordict)  # Ok
```
</details>